lib/Dialect/AIE/IR/AIEDialect.cpp

```diff
@@ -2240,6 +2271,22 @@ LogicalResult BDChainOp::verify() {
   return success();
 }
 
+//===----------------------------------------------------------------------===//
+// ShimDMAAllocationOp
+//===----------------------------------------------------------------------===//
+
+ShimDMAAllocationOp ShimDMAAllocationOp::getForSymbol(DeviceOp device,
+                                                      llvm::StringRef symbol) {
+  auto alloc_ops = device.getOps<ShimDMAAllocationOp>();
+  for (auto it = alloc_ops.begin(); it != alloc_ops.end(); ++it) {
+    AIE::ShimDMAAllocationOp a = *it;
+    if (a.getSymName() == symbol) {
+      return a;
+    }
+  }
```

Comment:
```c++
  for (auto allocOp : device.getOps<ShimDMAAllocationOp>())
    if (allocOp.getSymName() == symbol)
      return allocOp;
```
I also suggest taking a look at `ShimDMAllocationGetter` in AIEDmaToNpu.cpp to determine if these can be combined. That version does a symbol lookup to find the op instead of iterating over all ops. In any case it was identified as a hot spot.

Reply:
Thanks for pointing to that. I tried using the `ShimDMAAllocationGetter`, however it seems to only work if the symbol is defined somewhere. The `shim_dma_allocation` op is not the op that defines the symbol, it is merely a user. I don't know what's actually supposed to define the symbol, but I suppose I could see from the objectfifo lowering.

Could this be ok as a future PR to clean that up or would you like to see it in this one?

Reply:
> Could this be ok as a future PR to clean that up or would you like to see it in this one?

Yeah future PR. I just wanted flag that there was a similar helper class elsewhere.